src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp

```diff
@@ -376,6 +376,16 @@ bool MakeString::convertImpl(String & out, IParser::Pos & pos)
 
 bool NewGuid::convertImpl(String & out, IParser::Pos & pos)
 {
+    const String fn_name = getKQLFunctionName(pos);
+    if (fn_name.empty())
+        return false;
+    ++pos;
+    ++pos;
+    if (!pos->isEnd() && pos->type != TokenType::Semicolon && pos->type != TokenType::PipeMark)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Function {} expects 0 argument(s)", fn_name);
+    --pos;
+    --pos;
+    --pos;
     return directMapping(out, pos, "generateUUIDv4");
```

Comment:
This should do the trick:
```suggestion
    return directMapping(out, pos, "generateUUIDv4", {0, 0});
```